cmd/webhooks/main.go

```diff
@@ -0,0 +1,180 @@
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	kedav1alpha1 "github.com/kedacore/keda/v2/apis/keda/v1alpha1"
+	"github.com/kedacore/keda/v2/pkg/k8s"
+	kedautil "github.com/kedacore/keda/v2/pkg/util"
+	"github.com/kedacore/keda/v2/version"
+	//+kubebuilder:scaffold:imports
+)
+
+var webhooks = []rotator.WebhookInfo{
+	{
+		Name: "keda-validating-webhooks",
+		Type: rotator.Validating,
+	},
+}
+
+var (
+	scheme         = apimachineryruntime.NewScheme()
+	setupLog       = ctrl.Log.WithName("setup")
+	secretName     = "kedaorg-webhooks-secret" // #nosec
+	serviceName    = "keda-webhooks"
+	caName         = "kedaorg-ca"
+	caOrganization = "kedaorg"
+	// DNSName is <service name>.<namespace>.svc
+	dnsName = fmt.Sprintf("%s.%s.svc", serviceName, kedautil.GetPodNamespace())
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+
+	utilruntime.Must(kedav1alpha1.AddToScheme(scheme))
+	//+kubebuilder:scaffold:scheme
+}
+
+func main() {
+	var metricsAddr string
+	var probeAddr string
+	var adapterClientRequestQPS float32
+	var adapterClientRequestBurst int
+	var webhookCertDir string
+	var disableCertRotation bool
+	var tlsMinVersion string
+	pflag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
+	pflag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	pflag.Float32Var(&adapterClientRequestQPS, "kube-api-qps", 20.0, "Set the QPS rate for throttling requests sent to the apiserver")
+	pflag.IntVar(&adapterClientRequestBurst, "kube-api-burst", 30, "Set the burst for throttling requests sent to the apiserver")
+	pflag.StringVar(&webhookCertDir, "webhook-cert-dir", "/certs", "Webhook certificates dir to use. Defaults to /certs")
+	pflag.BoolVar(&disableCertRotation, "disable-cert-rotation", false, "disable automatic generation and rotation of webhook TLS certificates/keys")
+	pflag.StringVar(&tlsMinVersion, "tls-min-version", "1.3", "Minimum TLS version")
+
+	opts := zap.Options{}
+	opts.BindFlags(flag.CommandLine)
+	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
+	pflag.Parse()
+
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+
+	cfg := ctrl.GetConfigOrDie()
+	cfg.QPS = adapterClientRequestQPS
+	cfg.Burst = adapterClientRequestBurst
+
+	mgr, err := ctrl.NewManager(cfg, ctrl.Options{
+		Scheme:                 scheme,
+		LeaderElection:         false,
+		MetricsBindAddress:     metricsAddr,
+		Port:                   9443,
+		HealthProbeBindAddress: probeAddr,
+		CertDir:                webhookCertDir,
+	})
+	if err != nil {
+		setupLog.Error(err, "unable to start webhooks")
+		os.Exit(1)
+	}
+
+	// Make sure certs are generated and valid if cert rotation is enabled.
+	setupFinished := make(chan struct{})
+	if !disableCertRotation {
```

Comment:
what if user would like to add own certs?

Reply:
They can do it. I'll update docs explaining how to do it, but it's just setting the argument `disable-cert-rotation: true` and placing the certificates in the correct path (`/certs` as default but configurable with the argument `webhook-cert-dir`)

Reply:
I have renamed the parameter to `enable-cert-rotation`. This parameter with ensure that the secret provided by `webhooks-cert-secret-name` exists and will place the certificates there. 
For scenarios with user-provided certificates, this parameter has to be removed or switched to false and also users have to ensure that valid certificates are placed in the folder given by `webhooks-cert-dir` (`/certs` as default).
I have created [an issue](https://github.com/kedacore/keda-docs/issues/1030) in docs to properly documented because I'd like to document also other certificates, all together in a specific section